src/estimagic/benchmarking/run_benchmark.py

```diff
@@ -154,40 +168,52 @@ def _get_kwargs_list_and_names(problems, opt_options):
     return kwargs_list, names
 
 
-def _get_results(names, raw_results, kwargs_list):
-    registry = get_registry(extended=True)
-    results = {}
-
-    for name, result, inputs in zip(names, raw_results, kwargs_list):
-        if isinstance(result, OptimizeResult):
-            history = result.history
-            params_history = pd.DataFrame(
-                [tree_just_flatten(p, registry=registry) for p in history["params"]]
-            )
-            criterion_history = pd.Series(history["criterion"])
-            batches_history = pd.Series(history["batches"])
-            time_history = pd.Series(history["runtime"])
-        elif isinstance(result, str):
-            _criterion = inputs["criterion"]
-
-            params_history = pd.DataFrame(
-                tree_just_flatten(inputs["params"], registry=registry)
-            ).T
-            criterion_history = pd.Series(_criterion(inputs["params"])["value"])
-
-            time_history = pd.Series([np.inf])
-            batches_history = pd.Series([0])
-        else:
-            raise TypeError(
-                "'result' object is expected to be of type 'dict' or 'str'."
-            )
-
-        results[name] = {
-            "params_history": params_history,
-            "criterion_history": criterion_history,
-            "time_history": time_history,
-            "batches_history": batches_history,
-            "solution": result,
-        }
+def _process_one_result(optimize_result, problem):
+    """Process the result of one optimization run.
 
-    return results
+    Args:
+        optimize_result (OptimizeResult): Result of one optimization run.
+        problem (dict): Problem specification.
+
+    Returns:
+        dict: Processed result.
+
+    """
+    _registry = get_registry(extended=True)
+    _criterion = problem["noise_free_criterion"]
+    _start_x = problem["inputs"]["params"]
+    _start_crit_value = _criterion(_start_x)
+    if isinstance(_start_crit_value, np.ndarray):
+        _start_crit_value = (_start_crit_value**2).sum()
```

Comment:
If `_start_crit_value` is a `np.ndarray` we are always in the least-squares case?